eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -762,49 +772,51 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       // otherwise we cannot be sure that it was correctly received before disconnecting.
       stay()
 
+    // Channels are publicly announced if both parties want it: we ignore this message if we don't want to announce the channel.
     case Event(remoteAnnSigs: AnnouncementSignatures, d: DATA_NORMAL) if d.commitments.announceChannel =>
-      // Channels are publicly announced if both parties want it: we ignore this message if we don't want to announce the channel.
-      d.shortIds.real_opt match {
-        case Some(realScid) if remoteAnnSigs.shortChannelId == realScid =>
-          d.channelAnnouncement match {
-            case None =>
-              log.info("announcing channelId={} on the network with shortChannelId={}", d.channelId, realScid)
-              // We already sent our announcement_signatures but we don't store them so we need to recompute them.
-              val localAnnSigs = Helpers.makeAnnouncementSignatures(nodeParams, d.commitments.params, d.commitments.latest.remoteFundingPubKey, realScid)
-              val fundingPubKey = keyManager.fundingPublicKey(d.commitments.params.localParams.fundingKeyPath, fundingTxIndex = 0) // TODO: public announcements are not yet supported with splices
-              val channelAnn = Announcements.makeChannelAnnouncement(nodeParams.chainHash, localAnnSigs.shortChannelId, nodeParams.nodeId, d.commitments.params.remoteParams.nodeId, fundingPubKey.publicKey, d.commitments.latest.remoteFundingPubKey, localAnnSigs.nodeSignature, remoteAnnSigs.nodeSignature, localAnnSigs.bitcoinSignature, remoteAnnSigs.bitcoinSignature)
+      d.commitments.resolveCommitment(remoteAnnSigs.shortChannelId) match {
+        case Some(c) if d.lastAnnouncedCommitment_opt.exists(_.fundingTxIndex > c.fundingTxIndex) =>
+          log.info("ignoring remote announcement_signatures for scid={} with fundingTxIndex={}, we have a more recent announcement with scid={} and fundingTxIndex={}", remoteAnnSigs.shortChannelId, c.fundingTxIndex, d.lastAnnouncement_opt.map(_.shortChannelId), d.lastAnnouncedCommitment_opt.map(_.fundingTxIndex))
+          stay()
+        case Some(c) if c.announcement_opt.nonEmpty =>
+          if (!announcementSigsSent.contains(remoteAnnSigs.shortChannelId)) {
+            log.info("re-sending announcement_signatures for scid={}", remoteAnnSigs.shortChannelId)
+            val localAnnSigs_opt = c.signAnnouncement(nodeParams, d.commitments.params)
+            localAnnSigs_opt.foreach(annSigs => announcementSigsSent += annSigs.shortChannelId)
+            stay() sending localAnnSigs_opt.toSeq
+          } else {
+            log.debug("ignoring remote announcement_signatures for scid={} with fundingTxIndex={}, channel is already announced", remoteAnnSigs.shortChannelId, c.fundingTxIndex)
```

Comment:
This log isn't entirely accurate:
```suggestion
            log.debug("ignoring remote announcement_signatures for scid={} with fundingTxIndex={}, channel is announced and local announcement_signatures  were only sent once", remoteAnnSigs.shortChannelId, c.fundingTxIndex)
```